### PR TITLE
feat(android): add `device.tap()` and `device.longPress()`.

### PR DESCRIPTION
**Pull request by gayablau**
## Description

- This pull request addresses the issue described here: #<?>

In this pull request, I have …

---

> _For features/enhancements:_
 - [ ] I have added/updated the relevant references in the [documentation](https://github.com/wix/Detox/tree/master/docs) files.

> _For API changes:_
 - [x] I have made the necessary changes in the [types index](https://github.com/wix/Detox/blob/master/detox/index.d.ts) file.

### CODE REVIEW

**Comment by asafkorem** [comment]
@gayablau please fix the lint errors and unit tests.

File: detox/src/devices/runtime/drivers/DeviceDriverBase.js [hunk 1]
```diff
@@ -51,6 +51,10 @@ class DeviceDriverBase {
     return '';
   }
 
+  async tap(point) {
+    return '';
+  }
+
```
Comment (asafkorem) [inline detox/src/devices/runtime/drivers/DeviceDriverBase.js:57]: add a similar method for `longPress(arg1, arg2)` (in `DeviceDriverBase.js`).

File: detox/test/e2e/06.device.test.js [hunk 2]
```diff
@@ -84,4 +84,35 @@ describe('Device', () => {
...
+    await expect(element(by.text('Screen Long Pressed'))).toBeVisible();
+  });
+
+  it(':android: long press on screen by coordinates with duration', async () => {
+    await device.launchApp({ newInstance: true });
+    await element(by.text('Device Tap')).tap();
+    const point = {x: 200, y: 190}
+    await device.longPress(point, 2000);
+    await expect(element(by.text('Screen Long Custom Duration Pressed'))).toBeVisible();
+  });
```
Comment (asafkorem) [inline detox/test/e2e/06.device.test.js:117]: let's extract these to a new suite (e.g. `06.device-tap.test.js`)

Comment (asafkorem) [reply detox/test/e2e/06.device.test.js:117]: also add e2e tests for long-press with (1) duration only and (2) no params

Comment (asafkorem) [reply detox/test/e2e/06.device.test.js:117]: I would also add more tests:
- `should long press on specified point` (expect for **specific area** to be pressed when passing the `point` param)
- `should not succeed in long pressing with point outside the target area`

Check how we added a similar tests on `03.actions.test.js`.

Comment (asafkorem) [reply detox/test/e2e/06.device.test.js:117]: Same for `device.tap(point)`

**Review by asafkorem — COMMENTED** [review]
Great work! 👏🏼 🤩 
We also need to [document](https://wix.github.io/Detox/docs/api/device) this (`device.md`).

File: detox/src/devices/runtime/drivers/android/AndroidDriver.js [hunk 3]
```diff
@@ -245,6 +245,20 @@ class AndroidDriver extends DeviceDriverBase {
...
   }
 
+  async tap(point) {
+    let x = point?.x || 100;
+    let y = point?.y || 100;
+    const call = EspressoDetoxApi.tap(x, y);
+    await this.invocationManager.execute(call);
+  }
+
+  async longPress(point, duration) {
```
Comment (asafkorem) [inline detox/src/devices/runtime/drivers/android/AndroidDriver.js:255]: please add the relevant unit tests (`androiddriver.test.js`)

**Review by asafkorem — CHANGES_REQUESTED** [review]

**Comment by asafkorem** [comment]
Also let's add assertion when calling this API on iOS (raise unsupported message) - we should have that until adding the support on iOS.

File: detox/test/e2e/06.device.test.js (hunk 2 above)
Comment (gayablau) [reply detox/test/e2e/06.device.test.js]: about the spetific point test these are `tap on button by coordinates ignoring status bar` and `long press on button by coordinates with duration ignoring status bar`. about failing on press outside the are these are `long press on screen with duration without specifying coordinates` and `tap on screen without specifying coordinates`
tell me if It is what you meant

**Comment by gayablau** [comment]
united android and ios branches. fixes of review pushed to https://github.com/wix/Detox/pull/4542

**Pull request closed without merging**